### PR TITLE
backport the zone concept to 1.8

### PR DESCRIPTION
**Pull request by xabbuh**

### CODE REVIEW

**Comment by Ener-Getick** [comment]
Weren't there any conflicts?

**Comment by xabbuh** [comment]
There were some conflicts regarding the order of use statements and the usage of the short array notation. But the feature itself didn't cause any conflicts.

File: Resources/doc/3-listener-support.rst [hunk 1]
```diff
@@ -191,12 +191,32 @@ You need to enable this listener as follows, as it is disabled by default:
 
 It is also recommended to enable the exception controller described in the next chapter.
 
+Zone Listener
+=============
+
+As you can see, FOSRestBundle provides multiple event listeners to enable REST-related features.
```
Comment (Ener-Getick) [inline Resources/doc/3-listener-support.rst]: Do we need the ``as you can see``?

**Comment by Ener-Getick** [comment]
Status: reviewed
:+1:

File: Resources/doc/3-listener-support.rst (hunk 1 above)
Comment (xabbuh) [inline Resources/doc/3-listener-support.rst]: Probably no, but I suggest to tweaks things in a separate PR.

**Pull request merged**